### PR TITLE
Add support for listing shared domains (/v2/shared_domains)

### PR DESCRIPTION
**Pull request by peterellisjones**

### CODE REVIEW

File: domains.go [hunk 1]
```diff
@@ -30,6 +42,14 @@ type Domain struct {
 	c                      *Client
 }
 
+type SharedDomain struct {
+	Guid            string `json:"guid"`
+	Name            string `json:"name"`
+	RouterGroupGuid string `json:"router_group_guid"`
+	RouteGroupType  string `json:"router_group_type"`
```
Comment (frodenas) [inline domains.go]: Small typo, it should be `RouterGroupType` (missing an `r`)

File: domains_test.go [hunk 2]
```diff
@@ -29,6 +29,28 @@ func TestListDomains(t *testing.T) {
 	})
 }
 
+func TestListSharedDomains(t *testing.T) {
+	Convey("List domains", t, func() {
```
Comment (frodenas) [inline domains_test.go]: List `shared` domains

File: payloads_test.go [hunk 3]
```diff
@@ -1745,6 +1745,92 @@ const listDomainsPayload = `{
   ]
 }`
 
+const listSharedDomainsPayload = `{
+  "total_results": 6,
```
Comment (frodenas) [inline payloads_test.go]: Can you please reduce the number of results? the `payloads_test.go` is getting bigger and bigger, so let's add just the tests that we really need.

**Comment by peterellisjones** [comment]
Thanks for the feedback @frodenas, I've fixed the issues you found.

I also found some misnamed method arguments from #71, so I've fixed that and added the changes to this PR (dc2a9e8cf7ff65d98b476a7b7452a0255d16ae61)

**Comment by frodenas** [comment]
Thanks @peterellisjones

**Pull request merged**